### DISCUSSION TITLE
auto-generate nav: object in mkdocs.yml

### DISCUSSION THREAD

**Discussion by bpgould** [mkdocs/mkdocs#2815] [Q&A]
I would like to pass a top level directory to mkdocs and have it build the navigation 1:1 with the directory structure in the repo. I could probably write a python script to use os.walk() or something similar and parse it to an mkdocs.yaml with pyYAML, but I would rather not. Is there anything like this in the pipeline?

**Comment by bpgould** [comment]
For clarification:
```
dir 1
   subject
        file.md
        file2.md
   new_subject
        file3.md

```
etc. etc.

I would like to have mkdocs:
nav:
    - "useDir": "dir1/"

**Reply by LeFouineur** [reply]
Hi,

If I understood correctly what you are trying to to, it's already the case.

First, you need to remove the **NAV** section of your **mkdocs.yml**.
That will allow MKDocs to autogenerate the Navigation hierarchy based on the folders structure.

For the Name of the sections, it will be based on the Title (**#**) inside the MD file or by the name of the file.

More at [How to layout and write your Markdown source files.](https://www.mkdocs.org/user-guide/writing-your-docs/)
